### ISSUE TITLE
[Bug]: CLI verify command fails for non-PyTorch profiles (TensorFlow/JAX

### ISSUE THREAD

**Issue by MayurKharat0390** [rishabh0510rishabh/EnvForage#503]
### Describe the Bug
The `envforge verify` command in the CLI agent is designed to verify whether an environment was correctly set up and check if the installed framework has access to graphics hardware (CUDA). 

Currently, both the subprocess verification script and the Rich console formatter hardcode PyTorch (`import torch`) checks:
1. The subprocess script only attempts to `import torch` and check `torch.cuda.is_available()`.
2. The print helper `_print_verification_summary` hardcodes labels like `"PyTorch Core Import"`.

If a developer generates and verifies a TensorFlow or JAX profile (such as `tensorflow-cuda` or `jax-cuda`), the `verify` command fails with a PyTorch import error even if TensorFlow or JAX was successfully set up with GPU support.

### Expected Behavior
The CLI `verify` command should:
1. Detect the target framework based on the `--profile` option (e.g., TensorFlow for `tensorflow-cuda`, JAX for `jax-cuda`, PyTorch by default).
2. Execute the corresponding verification imports and GPU checks:
   * **TensorFlow:** `import tensorflow`, check `tf.config.list_physical_devices('GPU')`.
   * **JAX:** `import jax`, check `jax.devices()` GPU platform presence.
   * **PyTorch:** `import torch`, check `torch.cuda.is_available()`.
3. Dynamically format the verification summary tables using the detected framework name.

### Steps to Reproduce
1. Set up an environment containing only `tensorflow` or `jax` (no `torch` installed).
2. Run the verification command:
   ```bash
   envforge verify --profile tensorflow-cuda

### To Reproduce

1. Set up a Python environment where TensorFlow or JAX is installed, but PyTorch (`torch`) is NOT installed.
2. Run the environment verification command specifying a non-PyTorch profile slug:
   envforge verify --profile tensorflow-cuda
3. The command crashes with a failure status stating "PyTorch import failed" because the verification script contains hardcoded PyTorch imports.

### Expected behavior

The verify command should dynamically inspect the `--profile` argument to detect the target framework (e.g. TensorFlow, JAX, or PyTorch). 

It should run framework-specific import and GPU checks (such as using tf.config.list_physical_devices('GPU') for TensorFlow, and jax.devices() for JAX), and format the check matrix table with the correct dynamic framework name.

### Diagnostic Report

```json
{
  "agent_version": "1.0.1",
  "os": {
    "name": "Ubuntu 22.04",
    "version": "22.04",
    "architecture": "x86_64"
  },
  "cpu": {
    "brand": "AMD Ryzen 7 7840HS",
    "cores": 8,
    "threads": 16
  },
  "ram": {
    "total_gb": 16.0,
    "available_gb": 10.5
  },
  "gpus": [
    {
      "name": "NVIDIA RTX 4060 Laptop GPU",
      "vram_gb": 8.0,
      "driver_version": "555.85"
    }
  ]
}
```

### EnvForge Version

1.0.1 (CLI Agent)

### Component

CLI Agent (envforge-agent)

**Comment by MayurKharat0390** [comment]
---

### 🏷️ Program Labels Request
Dear maintainers, could you please label this issue under **NSoC'26**, **SSoC26**, and **gssoc-26**? I am participating in these summer of code programs and would love to be assigned to this issue to submit the PR for it. Thank you!

**Comment by rishabh0510rishabh** [comment]
assigend @MayurKharat0390 ,
